appinventor/components/src/com/google/appinventor/components/common/ScreenOrientation.java

```diff
@@ -0,0 +1,49 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2020 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package com.google.appinventor.components.common;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Defines a ScreenOrientation type used by the Form component to specify orientation.
+ */
+public enum ScreenOrientation implements OptionList<String> {
+  Unspecified("unspecified"),
```

Comment:
It would be interesting if in addition to encoding the string value used in the blocks to also encode the integer constant that is passed to `setRequestedOrientation` call. This would eliminate the need for the switch statement. Possible name for the getter is `getOrientationConstant()`.

The one problem is that I don't think the component common package is compiled with android.jar on the classpath, so you may need to add it in build.xml to make it compile.

Reply:
Ok I was a bit confused. I'm not sure if you wanted me to add the actual constants as documented [here](https://developer.android.com/reference/android/content/pm/ActivityInfo#SCREEN_ORIENTATION_BEHIND). Or if you wanted me to add calls to ActivityInfo.X

I decided to add the raw constants because I assume that evaluating any of the gingerbread values on a pre-gingerbread device would cause an error. As such we would need to hard code those constants anyway. So I figured hard coding the rest wouldn't be a problem.